lte/gateway/c/oai/tasks/sgw_s8/sgw_s8_handlers.c

```diff
@@ -1108,3 +1109,19 @@ void sgw_s8_handle_release_access_bearers_request(
   }
   OAILOG_FUNC_OUT(LOG_SGW_S8);
 }
+
+/* Generates random control plane teid and is used for both s11 and
+ * s8 interface to uniquely identify sgw_s8_context for pdn session
+ */
+static teid_t sgw_s8_generate_new_cp_teid(void) {
+  OAILOG_FUNC_IN(LOG_SGW_S8);
+  sgw_eps_bearer_context_information_t* sgw_context_p = NULL;
+  teid_t teid                                         = INVALID_TEID;
+  // note srand with seed is init at main
+  do {
+    teid          = (teid_t) rand();
+    sgw_context_p = sgw_get_sgw_eps_bearer_context(teid);
+  } while (sgw_context_p);
+
```

Comment:
does this ensure that TEID is unique across S8 and spgw tasks?

Reply:
No. This logic creates unique control plane tied in sgw_s8 task. The same teid is assigned to both s11 and s8 interface as it maps to single PDN session. 
But we don't need to have unique tied across s8 and spgw tasks. Because MME may be connected with multiple SGWs but we don't  need have unique control plane teids across SGWs.
In mme, sgw_s11_cp_teid is updated per PDN, so I think it should be fine
Me and @uri200  had discussion and Oriol modified inbound_roaming_02.ntl script to run both roaming and non-roaming subscribers and I generated mme binary with same teid assigned to both S8 task and spgw task. 
It worked fine.

A unique sgw_s11_teid is required in SGW to identify procedure belongs to specific PDN and later to identify specific bearer, we use the ebi.
Is there any other requirement?

Reply:
Okay good points.